src/Jackalope/Node.php

```diff
@@ -213,9 +221,9 @@ private function parseData($rawData, $update, $keepChanges = false)
                     continue;
                 }
 
-                if ($update && array_key_exists($key, $this->properties)) {
-                    unset($oldProperties[$key]);
-                    $prop = $this->properties[$key];
+                if ($update && array_key_exists($key, $this->propertyValues)) {
+                    unset($oldPropertyValues[$key]);
+                    $prop = $this->loadProperty($key);
```

Comment:
i think same here, this is only about properties that are already loaded. if they are not loaded, they can't be modified or deleted (or does deleting a property using Node::setValue not lead to the property being loaded?)